### DISCUSSION TITLE
Access real and imaginary parts of mx.complex64

### DISCUSSION THREAD

**Discussion by louisamand** [ml-explore/mlx#1331] [Q&A] [answered]
It seems it is not yet implemented (https://github.com/ml-explore/mlx/issues/361), but is there a workaround that doesn't force a conversion to NumPy ?

**Comment by angeloskath** [comment, marked as answer]
We should implement that for sure! In the meantime the following should work well

```python
def real(x):
    return mx.view(x, mx.float32)[..., ::2]


def imag(x):
    return mx.view(x, mx.float32)[..., 1::2]
```

**Reply by louisamand** [reply]
Thanks! quite gracious :)
